x-pack/plugins/enterprise_search/public/applications/app_search/components/documents/documents_detail_logic.test.ts

```diff
@@ -0,0 +1,156 @@
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { resetContext } from 'kea';
+
+import { mockHttpValues } from '../../../__mocks__';
+jest.mock('../../../shared/http', () => ({
+  HttpLogic: { values: mockHttpValues },
+}));
+const { http } = mockHttpValues;
+
+jest.mock('../engine', () => ({
+  EngineLogic: { values: { engineName: 'engine1' } },
+}));
+
+jest.mock('../../../shared/flash_messages', () => ({
+  setQueuedSuccessMessage: jest.fn(),
+  flashAPIErrors: jest.fn(),
+}));
+import { setQueuedSuccessMessage, flashAPIErrors } from '../../../shared/flash_messages';
+
+import { DocumentDetailLogic, DocumentDetailLogicValues } from './document_detail_logic';
+
+describe('DocumentsLogic', () => {
+  const DEFAULT_VALUES: DocumentDetailLogicValues = {
+    dataLoading: true,
+    fields: [],
+  };
+
+  const mount = (defaults?: object) => {
+    if (!defaults) {
+      resetContext({});
+    } else {
+      resetContext({
+        defaults: {
+          enterprise_search: {
+            app_search: {
+              document_detail_logic: {
+                ...defaults,
+              },
+            },
+          },
+        },
+      });
+    }
+    DocumentDetailLogic.mount();
+  };
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('actions', () => {
+    describe('setFields', () => {
+      it('should set fields to the provided value and dataLoading to false', () => {
+        const fields = [{ name: 'name', value: 'python', type: 'string' }];
+
+        mount({
+          dataLoading: true,
+          fields: [],
+        });
+
+        DocumentDetailLogic.actions.setFields(fields);
+
+        expect(DocumentDetailLogic.values).toEqual({
+          ...DEFAULT_VALUES,
+          dataLoading: false,
+          fields,
+        });
+      });
+    });
+
+    describe('getDocumentDetails', () => {
+      it('will call an API endpoint and then store the result', async () => {
+        const fields = [{ name: 'name', value: 'python', type: 'string' }];
+        jest.spyOn(DocumentDetailLogic.actions, 'setFields');
+        const promise = Promise.resolve({ fields });
+        http.get.mockReturnValue(promise);
+
+        DocumentDetailLogic.actions.getDocumentDetails('1');
+
+        expect(http.get).toHaveBeenCalledWith(`/api/app_search/engines/engine1/documents/1`);
+        await promise;
+        expect(DocumentDetailLogic.actions.setFields).toHaveBeenCalledWith(fields);
+      });
+
+      it('handles errors', async () => {
+        mount();
+        const promise = Promise.reject('An error occured');
+        http.get.mockReturnValue(promise);
+
+        try {
+          DocumentDetailLogic.actions.getDocumentDetails('1');
+          await promise;
+        } catch {
+          // Do nothing
+        }
+        expect(flashAPIErrors).toHaveBeenCalledWith('An error occured');
```

Comment:
[nit] same as above, feel free to ignore
```suggestion
        expect(flashAPIErrors).toHaveBeenCalledWith('An error occurred');
```